CoreBluetoothMock/CBMPeripheralSpecDelegate.swift

```diff
@@ -251,155 +251,113 @@ public protocol CBMPeripheralSpecDelegate {
+            
+        peripheral(p, didReceiveCharacteristicsDiscoveryRequest: characteristicUUIDs, for: service)
     }
     
-    func peripheral(_ peripheral: CBMPeripheralSpec,
-                    didReceiveDescriptorsDiscoveryRequestFor characteristic: CBMCharacteristic)
-        -> Result<Void, Error> {
-            return .success(())
+    func peripheral(_ peripheral: CBMPeripheralSpec, didReceiveDescriptorsDiscoveryRequestFor characteristic: CBMCharacteristic) -> Result<Void, Error> {
+        .success(())
     }
     
-    func peripheral(_ p: CBMPeripheralSpec,
-                    didReceiveDescriptorsDiscoveryRequestFor characteristic: CBMCharacteristicMock)
-        -> Result<Void, Error> {
-            return peripheral(p, didReceiveDescriptorsDiscoveryRequestFor: characteristic as CBMCharacteristic)
+    func peripheral(_ p: CBMPeripheralSpec, didReceiveDescriptorsDiscoveryRequestFor characteristic: CBMCharacteristicMock) -> Result<Void, Error> {
+            
+        peripheral(p, didReceiveDescriptorsDiscoveryRequestFor: characteristic)
     }
     
-    func peripheral(_ peripheral: CBMPeripheralSpec,
-                    didReceiveReadRequestFor characteristic: CBMCharacteristic)
-        -> Result<Data, Error> {
-            return .failure(CBMATTError(.readNotPermitted))
+    func peripheral(_ peripheral: CBMPeripheralSpec, didReceiveReadRequestFor characteristic: CBMCharacteristic) -> Result<Data, Error> {
+            
+        .failure(CBMATTError(.readNotPermitted))
     }
     
-    func peripheral(_ p: CBMPeripheralSpec,
-                    didReceiveReadRequestFor characteristic: CBMCharacteristicMock)
-        -> Result<Data, Error> {
-            return peripheral(p, didReceiveReadRequestFor: characteristic as CBMCharacteristic)
+    func peripheral(_ p: CBMPeripheralSpec, didReceiveReadRequestFor characteristic: CBMCharacteristicMock) -> Result<Data, Error> {
+            
+        peripheral(p, didReceiveReadRequestFor: characteristic)
     }
     
-    func peripheral(_ peripheral: CBMPeripheralSpec,
-                    didReceiveReadRequestFor descriptor: CBMDescriptor)
-        -> Result<Data, Error> {
-            return .failure(CBMATTError(.readNotPermitted))
+    func peripheral(_ peripheral: CBMPeripheralSpec, didReceiveReadRequestFor descriptor: CBMDescriptor) -> Result<Data, Error> {
+            
+        .failure(CBMATTError(.readNotPermitted))
     }
     
-    func peripheral(_ p: CBMPeripheralSpec,
-                    didReceiveReadRequestFor descriptor: CBMDescriptorMock)
-        -> Result<Data, Error> {
-            return peripheral(p, didReceiveReadRequestFor: descriptor as CBMDescriptor)
+    func peripheral(_ p: CBMPeripheralSpec, didReceiveReadRequestFor descriptor: CBMDescriptorMock) -> Result<Data, Error> {
+            
+        peripheral(p, didReceiveReadRequestFor: descriptor)
```

Comment:
Won't removing `as CBMDescriptor` cause this to call the same method in recursion over and over, instead of the one above with `CBMDescriptor` parameter type?

Reply:
This "deprecation" warning isn't correct. This is to make sure the "new" new method calls "deprecated" method if it's not implemented just in case someone has implemented the old one. This is normal thing when supporting legacy deprecated code, which internally calls deprecated methods for backwards compat. It's Xcode bug, not the library.